samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/ExtendedTimeQueries.java

```diff
@@ -157,4 +157,19 @@ private ExtendedTime getPubAndX(final String query, final PublishedAssessmentIfc
             return null;
         }
     }
+
+    public ExtendedTime getEntry(final String entryId){
+        log.debug("getEntry " + entryId);
+        try {
+            HibernateCallback hcb = (Session s) -> {
+                Query q = s.getNamedQuery(QUERY_GET_ENTRY);
+                q.setParameter(ENTRY_ID, Long.valueOf(entryId));
+                return q.uniqueResult();
```

Comment:
this queries for an entity using it's id which is a primary key and therefore uniqueness is enforced by the db so calling uniqueResult() is acceptable